processor/resourcedetectionprocessor/internal/aws/eks/detector.go

```diff
@@ -114,3 +129,44 @@ func (e eksDetectorUtils) getConfigMap(ctx context.Context, namespace string, na
 	}
 	return cm.Data, nil
 }
+
+func (e eksDetectorUtils) getClusterName() (string, error) {
+	cfg, err := config.LoadDefaultConfig(context.TODO())
+	if err != nil {
+		return "", err
+	}
+
+	ec2IMS := imds.NewFromConfig(cfg)
+	regionOutput, err := ec2IMS.GetRegion(context.TODO(), &imds.GetRegionInput{})
+	if err != nil {
+		return "", err
+	}
+	cfg.Region = regionOutput.Region
+
+	client := ec2.NewFromConfig(cfg)
+	describeInstances, err := client.DescribeInstances(context.TODO(), &ec2.DescribeInstancesInput{})
```

Comment:
I think this will query all ec2 instances, and we should get the tags for the local instance only. We can get the instance-id from imds and pass the instance id to `DescribeInstancesInput`.

Reply:
Done.